oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/Commit.java

```diff
@@ -354,6 +358,39 @@ private void applyToDocumentStore(RevisionVector baseBranchRevision)
         boolean commitRootHasChanges = operations.containsKey(commitRootPath);
         for (UpdateOp op : operations.values()) {
             NodeDocument.setCommitRoot(op, revision, commitRootDepth);
+
+            // special case for :childOrder updates
+            if (nodeStore.isChildOrderCleanupEnabled()) {
+                final Branch localBranch = getBranch();
+                if (localBranch != null) {
+                    final NavigableSet<Revision> commits = new TreeSet<>(localBranch.getCommits());
+                    boolean removePreviousSetOperations = false;
+                    for (Map.Entry<Key, Operation> change : op.getChanges().entrySet()) {
+                        if (":childOrder".equals(change.getKey().getName()) && Operation.Type.SET_MAP_ENTRY == change.getValue().type) {
```

Comment:
Ideally I guess there would be some oak-wide static for `:childOrder` - but as it breaks bundle barriers that might be too much effort. Wondering if we should have a static in `Commit` at least rather than having it in this method twice?